src/bit-fields-macros/src/lib.rs

```diff
@@ -0,0 +1,1065 @@
+                field_ident = field_ident
+            )
+            .unwrap();
+        };
+        // To check whether the field is a bit flag or bit field we check if the next token is `.`
+        // (which indicates a range)
+        #[allow(clippy::pattern_type_mismatch)]
+        match fields_iter.peek() {
+            // If we have `<ident>,<punctuation>,<literal>,<.>` it is a bit range
+            Some(TokenTree::Punct(punct)) if punct.as_char() == '.' => {
+                // Skip what we already checked by peeking
+                fields_iter.next();
+                /// Bit range correct punctuation description.
+                const CORRECT_BIT_RANGE_PUNCTUATION: &str = "Assuming this is the definition of a \
+                                                             bit range, this token should be the \
+                                                             punctuation character '.'.";
+                // Check 2nd '.' in bit range (e.g. 2nd dot/3rd character here `1..2`)
+                match fields_iter.next() {
+                    Some(TokenTree::Punct(punct)) if punct.as_char() == '.' => {}
+                    Some(TokenTree::Punct(punct)) => macro_error!(
+                        punct,
+                        format!(
+                            "Incorrect bit range punctuation. {}",
+                            CORRECT_BIT_RANGE_PUNCTUATION
+                        )
+                    ),
+                    Some(token_tree) => macro_error!(
+                        token_tree,
+                        format!(
+                            "Incorrect bit range token. {}",
+                            CORRECT_BIT_RANGE_PUNCTUATION
+                        )
+                    ),
+                    None => {
+                        macro_error!(
+                            field_ident,
+                            format!("Missing bit range token. {}", CORRECT_BIT_RANGE_PUNCTUATION)
+                        )
+                    }
+                };
+                // Check and get end index in bit range (e.g. 2nd literal/4th character here `1..2`)
+                let end = match fields_iter.next() {
+                    Some(TokenTree::Literal(field_end_pos)) => {
+                        if let Ok(pos) = field_end_pos.to_string().parse::<u8>() {
+                            pos
+                        } else {
+                            macro_error!(
+                                field_end_pos,
+                                "Failed to parse token for bit range end index."
+                            );
+                        }
+                    }
+                    Some(token_tree) => {
+                        macro_error!(token_tree, "Incorrect token for bit range end index.")
+                    }
+                    None => macro_error!(field_ident, "Missing token for bit range end index."),
+                };
+
+                let (valid_start, valid_end) = (
```

Comment:
Do we also maybe want to check if the range was not used before?

Reply:
When defining `start` we check `> pos` and whenever a range or flag is used `pos` is set to its end, thus we know `start` is outside any previously used range.